sparkle/Components/smac-v2.10.03-master-778/smac_target_algorithm.py

```diff
@@ -43,9 +46,15 @@
         if isinstance(quality, dict):
             # SMAC2 does not support multi-objective so always opt for the first value
             quality = quality[quality.keys()[0]]
+        elif isinstance(quality, list):
+            quality = quality[0]
         if not isinstance(quality, int) and not isinstance(quality, float):
             if not quality.replace("-", "").replace(".", "").isdigit():
                 # Not a number
                 quality = "0"
+    # SMAC2 does not do maximisation, auto-convert
+    if (objectives[0].PerformanceMeasure
+            == PerformanceMeasure.QUALITY_ABSOLUTE_MAXIMISATION):
+        quality = -1 * float(quality)
```

Comment:
This is a nice feature, however I would remove it here and add to another ticket that structurally enables to run sparkle for maximisation.